streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -938,11 +943,13 @@ public void close() {}
 
         final KafkaStreams streams = new KafkaStreams(builder.build(), config, new TestKafkaClientSupplier());
         streams.setUncaughtExceptionHandler((t, e) -> {
-            if (uncaughtException != null) {
+            // should only have our injected exception or commit exception
+            if (!(e instanceof RuntimeException) && !(e.getMessage().contains("test exception"))) {
+                // The exception won't cause the test fail since we actually "expected" exception thrown and failed the stream.
+                // So, log to stderr for debugging when the exception is not what we expected
                 e.printStackTrace(System.err);
                 fail("Should only get one uncaught exception from Streams.");
```

Comment:
> Seems this fail did not work as expected? Otherwise the test would have failed all the time? 
Yes, the `fail()` is throwing `AssertionError` and fail the stream, but that's excatly what we expected to fail the stream, just not with our injected exception.

> we need to handle this differently for the error-injection and the "clean run" differently depending on the boolean test flag.
I see, will update it. Thanks.

Reply:
Well, but the `AsseriontError` does not mark the test as failed as it seems, because it would be thrown in a different thread? We would need to set a boolean flag, and add an `assertThat` to the main test code to see if the flag fired?